discovery-provider/src/premium_content/premium_content_access_checker.py

```diff
@@ -0,0 +1,104 @@
+        if premium_content_type != "track":
+            return False, True
+
+        (
+            is_premium,
+            premium_conditions,
+            content_owner_id,
+        ) = self._is_content_premium(premium_content_id)
+
+        if not is_premium:
+            # premium_conditions should always be null here as it makes
+            # no sense to have a non-premium track with conditions
+            if premium_conditions:
+                logger.warn(
+                    f"premium_content_access_checker.py | _aggregate_conditions | non-premium content with id {premium_content_id} and type {premium_content_type} has premium conditions."
+                )
+            return False, True
+
+        if not premium_conditions:
+            # is_premium should always be false here as it makes
+            # no sense to have a premium track with no conditions
+            if is_premium:
+                logger.warn(
+                    f"premium_content_access_checker.py | _aggregate_conditions | premium content with id {premium_content_id} and type {premium_content_type} has no premium conditions."
+                )
+            return is_premium, True
+
+        does_user_have_access = self._evaluate_conditions(
+            user_id, cast(int, content_owner_id), premium_conditions
+        )
+        return True, does_user_have_access
+
+    # Returns a tuple of (bool, Dict | None, int | None) -> (track is premium, track premium conditions, track owner id)
+    def _is_content_premium(
+        self, premium_content_id: int
+    ) -> Tuple[bool, Optional[Dict], Optional[int]]:
+        db = db_session.get_db_read_replica()
+        with db.scoped_session() as session:
+            track = (
+                session.query(Track)
+                .filter(
+                    Track.track_id == premium_content_id,
+                    Track.is_current == True,
+                    Track.is_delete == False,
+                )
+                .first()
+            )
+
+            if not track:
+                return False, None, None
+
+            return track.is_premium, track.premium_conditions, track.owner_id
+
+    # There will eventually be another step prior to this one where
+    # we aggregate multiple conditions and evaluate them altogether.
+    # For now, we only support one condition, which is the ownership
+    # of an NFT from a given collection.
+    def _evaluate_conditions(
+        self, user_id: int, track_owner_id: int, premium_conditions: Dict
+    ):
+        if len(premium_conditions) != 1:
+            logging.info(
+                f"premium_content_access_checker.py | _aggregate_conditions | invalid conditions: {json.dumps(premium_conditions)}"
+            )
+            return False
+
+        condition, value = list(premium_conditions.items())[0]
+        if condition != "nft-collection":
+            return False
+
+        return does_user_have_nft_collection(user_id, value)
```

Comment:
I imagine at some point we'll want to have a map of access_condition -> access_condition_checker, but makes sense to keep this simple for now 👍

Reply:
yup! this is the vision :)